asapdiscovery-docking/asapdiscovery/docking/schema/pose_generation.py

```diff
@@ -692,6 +696,40 @@ def _generate_pose(
 
         return target_ligand
 
+    def poser(self, target_ligand, core_ligand, core_smarts, allowed_max_ha=75):
+        """
+        Generates a pose for a target_ligand while wrapping the whole thing in a try-except
+        so we can catch edge-cases. This enables multi-processed pose generation to return
+        exceptions gracefully.
+
+        Also skips molecules that are larger than `allowed_max_ha` which has been set to a
+        reasonable default. If erroneously large ligands are fed into this docking workflow
+        they will take a prohibitive amount of walltime because of the large number of embeddings
+        to enumerate.
+
+        Returns a success bool, the posed ligand (or input ligand in case of fail) and the
+        error message.
+        """
+        if len(target_ligand.smiles) > allowed_max_ha:
```

Comment:
This isn't the number of heavy atoms as includes Hs and stereochem descriptors no?